### ISSUE TITLE
Remove `util` from `aws-sdk`

### ISSUE THREAD

**Issue by benmccann** [43081j/ecosystem-cleanup#61]
I'd guess there could be a lighter-weight way of doing this

https://github.com/aws/aws-sdk-js/blob/5fedff2d17e3fa27f32489e2ee3fc97bbb2b0a76/package.json#L51

```
./lib/event_listeners.js:        var params = require('util').inspect(censoredParams, true, null);
./lib/node_loader.js:util.getSystemErrorName = require('util').getSystemErrorName;
```

**Comment by 43081j** [comment]
i agree. originally for these kinds of things, i was looking for a `util` alternative. however, i think most uses can probably just use much simpler logic than needing a full `util` polyfill

**Comment by AbhiPrasad** [comment]
aws-sdk v2 will be EOL september 2025, so I don't think they have much incentive to accept PRs for this kind of stuff.

https://github.com/aws/aws-sdk-js-v3 seems to be better, though the module graph for some AWS products is pretty [crazy](https://npmgraph.js.org/?q=%40aws-sdk%2Fclient-dynamodb).

**Comment by benmccann** [comment]
ooh, that's good to know. and wow, yeah. s3 is even crazier for something I thought would just need an http client: https://npmgraph.js.org/?q=@aws-sdk/client-s3

They're pretty much all internal packages, so I guess it's not as terrible. They've got a few packages that have tons of dependencies:
https://cdn.jsdelivr.net/npm/@aws-sdk/client-sts@3.606.0/package.json
https://cdn.jsdelivr.net/npm/@aws-sdk/client-sso-oidc@3.606.0/package.json

And they've got `node` and `browser` versions of packages rather than using `exports` conditions, which doubles the count:
`@smithy/util-body-length-browser`
`@smithy/util-body-length-node`

Anyway, since `util` has been remove from the v3, I'll go ahead and close this
